### ISSUE TITLE
Ugly CLI output when it exceeds terminal's height

### ISSUE THREAD

**Issue by MikeYermolayev** [docker/compose#9377]
**Description**

This issue is based on the discussion https://forums.docker.com/t/docker-compose-always-appends-to-console-output-instead-of-updating-it/122841/3

When the expected output exceeds my terminal's height then it's very hard to follow it since every update gets appended instead of replacing.

**Steps to reproduce the issue:**
1. Have docker-compose file with big amount of services (mine has around 30)
2. Run `docker-compose up` in terminal. 

**Describe the results you received:**
Every update to stdout gets appended instead of replacing current one

**Describe the results you expected:**
Update should replace current output so it's easy to understand the progress of operations

**Additional information you deem important (e.g. issue happens only occasionally):**
issue is happening only when terminal's height is lower than expected output height

**Output of `docker compose version`:**

```
Docker Compose version v2.3.3
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.8.1)
  compose: Docker Compose (Docker Inc., v2.3.3)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 63
  Running: 44
  Paused: 0
  Stopped: 19
 Images: 166
 Server Version: 20.10.13
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2a1d4dbdb2a1030dc5b01e96fb110a9d9f150ecc
 runc version: v1.0.3-0-gf46b6ba
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.104-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 3.842GiB
 Name: docker-desktop
 ID: UPSR:YQYB:DXRS:HJQC:XUY5:XUER:BH45:USMR:7NPK:CXX2:SVR4:7MRC
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 449
  Goroutines: 320
  System Time: 2022-04-11T11:32:03.108584552Z
  EventsListeners: 5
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details:**

For terminal I’m using vscode’s integrated terminal and zsh:

```
VsCode Version: 1.65.2
zsh 5.8 (x86_64-apple-darwin21.0)
```

**Comment by BBaoVanC** [comment]
This is still an issue
